src/ast/nodes/shared/FunctionNode.ts

```diff
@@ -58,12 +58,12 @@ export default class FunctionNode extends FunctionBase {
 		interaction: NodeInteraction,
 		context: HasEffectsContext
 	): boolean {
-		if (super.hasEffectsOnInteractionAtPath(path, interaction, context)) return true;
```

Comment:
Same comment as above.

Reply:
I think a better place might be to address this in `FunctionBase`, skipping everything below the first if-statement.